acceptance/tests/harvest_latest_versions.py

```diff
@@ -0,0 +1,175 @@
+# This script harvests the latest versions of the sources listed in a catalog.
+# Made for Python 3.9. Requires the modules listed in requirements.txt.
+# Copyright 2021 MobilityData IO.
+###############################################################################
+
+# API constants
+API_URL = "http://mobilitydatabase.org/w/api.php?"
+ACTION = "action"
+ARCHIVES_ID_PROPERTY = "P33"
+CLAIMS = "claims"
+ENTITIES = "entities"
+IDS = "ids"
+MODIFIED = "modified"
+SOURCE_PROPERTY = "P5"
+WBGETENTITIES = "wbgetentities"
+FORMAT = "format"
+JSON = "json"
+MAINSNAK = "mainsnak"
+DATAVALUE = "datavalue"
+VALUE = "value"
+ID = "id"
+
+# Catalog constants
+GTFS_CATALOG_ID = "Q6"
+
+# Archives ids file constants
+ARCHIVES_IDS = "archives_ids"
+HARVESTING_DATE = "harvesting_date"
+EPOCH_DATE = "1970-01-01T00:00:00Z"
+
+# Google Cloud constants
+LATEST_BUCKET_PATH = "{source_archives_id}_latest"
+LATEST_URL = "https://storage.googleapis.com/storage/v1/b/{source_archives_id}_latest/o/{blob_name}?alt=media"
+
+
+def parse_archives_ids_file(data_path, filename):
+    file_path = path.join(data_path, filename)
+    if path.exists(file_path):
+        with open(file_path, "r") as f:
+            archives_ids_file = json.load(f)
+        harvesting_date = archives_ids_file[HARVESTING_DATE]
+        archives_ids = archives_ids_file[ARCHIVES_IDS]
+    else:
+        harvesting_date = EPOCH_DATE
+        archives_ids = None
+    return harvesting_date, archives_ids
+
+
+def save_content_to_file(content, data_path, filename):
+    file_path = path.join(data_path, filename)
+    with open(file_path, "w") as f:
+        json.dump(content, f)
+
+
+def save_archives_ids_file(harvesting_date, archives_ids, data_path, filename):
+    archives_ids_json = {HARVESTING_DATE: harvesting_date, ARCHIVES_IDS: archives_ids}
+    save_content_to_file(archives_ids_json, data_path, filename)
+
+
+def get_entity_data(entity_id):
+    query = {ACTION: WBGETENTITIES, IDS: entity_id, FORMAT: JSON}
+    response = requests.get(API_URL, params=query)
+    response_json = response.json()
+    return response_json[ENTITIES][entity_id]
+
+
+def has_been_modified_since(catalog_data, last_harvesting_date):
+    modification_date = catalog_data[MODIFIED]
+    last_harvesting_date = dp.parse(last_harvesting_date)
+    modification_date = dp.parse(modification_date)
+    return modification_date > last_harvesting_date
+
+
+def harvest_archives_ids(catalog_data):
+    harvesting_date = catalog_data[MODIFIED]
+    archives_ids = []
+
+    sources = catalog_data[CLAIMS][SOURCE_PROPERTY]
+    for source in sources:
+        source_id = source[MAINSNAK][DATAVALUE][VALUE][ID]
+        source_data = get_entity_data(source_id)
+        source_archives_id = source_data[CLAIMS][ARCHIVES_ID_PROPERTY][0][MAINSNAK][DATAVALUE][VALUE]
+        archives_ids.append(source_archives_id)
+
+    return harvesting_date, archives_ids
+
+
+def harvest_latest_versions(archives_ids):
```

Comment:
```suggestion
def harvest_latest_versions(archives_ids, credentials):
```

Reply:
I will change this, but never do a `json.loads()` in a signature please 😅